framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -412,14 +430,75 @@ public ValueQualifierHierarchy(MultiGraphQualifierHierarchy.MultiGraphFactory fa
             super(factory);
         }
 
+        /**
```

Comment:
I still don't see a comment either here or on `greatestLowerBound` where this is called that explains why StringVal needs a more precise GLB than the other Value Checker types. Please include one.